### ISSUE TITLE
An error was reported in the last round of training

### ISSUE THREAD

**Issue by 2087987811** [harborsarah/GET-UP#8]
Dear author, why do I always generate complex (complex) type parameters or key points during the model training process, while the optimizer AdamW expects real (floating-point) type parameters?  And each time it is in the last round of setting the number of rounds，Sorry to disturb you. Looking forward to your reply 

<img width="1440" height="250" alt="Image" src="https://github.com/user-attachments/assets/47319534-9595-48e5-b311-ed50a2076c11" />

<img width="980" height="226" alt="Image" src="https://github.com/user-attachments/assets/28f4e5b6-56c9-46fe-a9fa-a59ede02a3c8" />

**Comment by harborsarah** [comment]
Hi, sorry, I've never met this error before. Could you maybe check where the parameters become complex numbers during training? That's a bit weird to me.

**Comment by 2087987811** [comment]
I'm not sure if my solution is correct, but from my investigation: the root cause isn't that the model parameters actually became complex numbers, but rather that complex results appeared in the learning rate update formula.

Before modification：
 for param_group in optimizer.param_groups:
                current_lr = (args.learning_rate - end_learning_rate) * (1 - global_step / num_total_steps) ** 0.9 + end_learning_rate
                param_group['lr'] = current_lr

After modification：
                for param_group in optimizer.param_groups:
                    progress = max(0.0, 1.0 - global_step / num_total_steps)  
                    current_lr = (args.learning_rate - end_learning_rate) * (progress ** 0.9) + end_learning_rate
                    param_group['lr'] = current_lr
My understanding is：
When the training step count global_step exceeds num_total_steps,
(1 - global_step / num_total_steps) becomes negative.
Raising a negative number to the power of 0.9 causes `current_lr` to become a complex number.

Additionally, I'm not sure if it's an issue with my environment's version. Overall, you've already provided very detailed instructions. Thank you very much.

While you're still here, I'd like to ask one more question: When I reproduced your work, I found that during training, the parameters corresponding to the best results are saved separately.
<img width="259" height="174" alt="Image" src="https://github.com/user-attachments/assets/e9d6f7e3-f733-469b-a0c7-20f4bc604d7e" />
So, in your thesis, the experimental results obtained correspond to the weighting file with the optimal parameter, or does each result correspond to a separate parameter file?
I apologize for the inconvenience. I look forward to your response.
